### DISCUSSION TITLE
Proposal for dpgen to use release-please action, making development automated and standardized

### DISCUSSION THREAD

**Discussion by saltball** [deepmodeling/dpgen#855] [General]
I suggest dpgen ( and maybe other deepmodeling projects ) to use release-please(https://github.com/googleapis/release-please) action as `dflow` do, especially for the version number bumping, CHANGELOG generation and breaking changing warnings.

`release-please` and [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) makes sure developers and users have clear commit history in the future. This action is used by some popular repos like [webssh2](https://github.com/billchurch/webssh2), [bootstrap-vue-3](https://github.com/cdmoro/bootstrap-vue-3), [protobuf.js](https://github.com/protobufjs/protobuf.js), [react-docgen](https://github.com/reactjs/react-docgen), [starship](https://github.com/starship/starship), [cloudquery](https://github.com/cloudquery/cloudquery), [npm/cli](https://github.com/npm/cli) and etc. See https://github.com/deepmodeling/dflow/issues/35 for details.

Please see [Release](https://github.com/deepmodeling/dflow/releases) and [CHANGELOG](https://github.com/deepmodeling/dflow/blob/master/CHANGELOG.md) of dflow generated by action-bot.

**Comment by wanghan-iapcm** [comment]
Looks interesting. @njzjz @AnguseZhang @HuangJiameng any comment?

**Comment by njzjz** [comment]
I don't think this action can be used for projects with different development and releasing branches.
